### PR TITLE
[FIX] account: global tax rounding error management

### PR DESCRIPTION
**Pull request by agr-odoo**
Create a sale order having these line values:
- Price Unit   54.45, tax 10% not included
- Price Unit  600.00, tax 10% not included
- Price Unit -500.00, tax 10% not included

Tax total will be:
|name|total|
|---|---|
|Untaxed Amount |154.45|
|Taxes     |      15.44|
|Total    |      169.89|

From the Sales Order create the Invoice

Issue:
Invoice tax totals will be
|name|total|
|---|---|
|Untaxed Amount |154.45|
|Taxes      |     15.45|
|Total       |   169.90|

This occurs because when managing the rounding errors for global
rounding the system 'force' a 1 cent difference on the negative line
that do not need rounding
Example:
Line 1
- current diff: 0.0
- computed tax amount: 5.455
- tax amount + diff: 5.455, after rounding: 5.45
- stored diff for next line: -0.005

Line 2
- current diff: -0.005
- computed tax amount: 60.0
- tax amount + diff: 59.995, after rounding: 60.0
- stored diff for next line: -0.005

Line 3
- current diff: -0.005
- computed tax amount: -50.0
- tax amount + diff: -50.005, after rounding: -50.01

A solution is to not add the diff when the tax line is already rounded

opw-3715229

### CODE REVIEW

File: addons/account/models/account_tax.py [hunk 1]
```diff
@@ -990,16 +990,18 @@ def add_tax_values(record, results, grouping_key, serialized_grouping_key, tax_v
...
-                                                     + total_amounts['delta_tax_amount_currency']
-                    tax_amount_currency = currency.round(tax_amount_currency_with_delta)
-                    tax_amount_with_delta = tax_values['tax_amount'] \
-                                            + total_amounts['delta_tax_amount']
-                    tax_amount = comp_currency.round(tax_amount_with_delta)
-                    tax_values['tax_amount_currency'] = tax_amount_currency
-                    tax_values['tax_amount'] = tax_amount
-                    total_amounts['delta_tax_amount_currency'] = tax_amount_currency_with_delta - tax_amount_currency
-                    total_amounts['delta_tax_amount'] = tax_amount_with_delta - tax_amount
+                    tax_amount_currency = tax_values['tax_amount_currency']
```
Comment (smetl) [inline addons/account/models/account_tax.py:993]: I removed the diff, the test is still passing.
Tested on the UI, a SO:
image

and the invoice:
image

Where is the issue exactly?

**Comment by smetl** [comment]
@agr-odoo I pushed on your branch, can you plz check it ? :)
Indeed, the test is run twice because the test in sale is extending the one in account without a "common" test. In this situation, the extended class is run first then the child making the test file run twice: one from the extension in sale, one from account :(

**Comment by agr-odoo** [comment]
@smetl 
Looks good, thanks!
I pushed to fix the complaining CI/style check and update the commit message

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by yeisonasuaje** [comment]
This is breaking when you don't have a pricelist on the sale order.

**Comment by randall-vx** [comment]
@hbto fyi

**Pull request closed without merging**